src/attitude_filter.cpp

```diff
@@ -28,58 +35,46 @@ const float error_measurement_gyro = 0.001;
 double t_current = 0.0;
 double t_last = 0.0;
 double dt = 0.0;
-int counter = 0;
-
-const Eigen::Matrix3d I = Eigen::Matrix3d::Identity();
-Eigen::MatrixXd A;
-const Eigen::MatrixXd B = Eigen::MatrixXd::Zero(15, 15);
-Eigen::MatrixXd C;
-
-//Initialising inital state
-const Eigen::VectorXd x0 = Eigen::VectorXd::Zero(9);
-const Eigen::Vector3d init_angles(x0(3), x0(4), x0(5));
-
-//Initialising error matrices
-Eigen::MatrixXd R;
-Eigen::VectorXd Q;
+bool first_step = true;
 
 //Initialization for kalman Update (global: no new initialization every step)
 Eigen::Vector3d current_angles;
 Eigen::VectorXd current_state;
-Eigen::VectorXd current_measurements;
 geometry_msgs::Quaternion current_orientation;
 
-//Initialising Kalman Filter
-KalmanFilter kalman(A, B, C, Q, R);
+//Declaration Kalman Filter
+KalmanFilter_Orientation kalman;
 
 int main(int argc, char** argv){
 
 	ros::init(argc, argv, "attitudefilter");
 	ros::NodeHandle node;
 
+	//Initialising state matrices
+	Eigen::Matrix<double,15,15> A = Eigen::MatrixXd::Identity(15,15);
```

Comment:
I guess he sets up the matrices and passes it to the kalman in its `init()` method down at row 77.
I would rather recommend doing so in the constructor of the kalman class if possible.
@SeleSchaefer google for "c++ member initializer list", it's the cleanest way to do this.